aten/src/ATen/native/cuda/CUDALoops.cuh

```diff
@@ -256,7 +256,7 @@ void gpu_kernel_impl_nocast(TensorIteratorBase& iter, const func_t& f) {
   TORCH_INTERNAL_ASSERT(iter.can_use_32bit_indexing());
   TORCH_INTERNAL_ASSERT(iter.ninputs() == traits::arity);
   TORCH_INTERNAL_ASSERT(iter.noutputs() == 1);
-  TORCH_INTERNAL_ASSERT(!needs_dynamic_casting<func_t>::check(iter));
+  TORCH_CHECK(!needs_dynamic_casting<func_t>::check(iter), "Dynamic casting is not currently supported on CPU");
```

Comment:
This is GPU kernel isn't it? And also, if someone calls `gpu_kernel_impl_nocast` with types that needs casting it indeed must raise an internal assert, as this function is not accessible as end user API

Reply:
You are totally correct!

Reply:
Restored the GPU API, thanks